idm/integrationpluginidm.cpp

```diff
@@ -0,0 +1,191 @@
+        Idm *idm = new Idm(hostAddress, this);
+        if (idm->connectDevice()) {
+            qCWarning(dcIdm()) << "Could not connect to thing";
+            info->finish(Thing::ThingErrorHardwareNotAvailable);
+            return;
+        }
+        m_idmConnections.insert(thing, idm);
+        connect(idm, &Idm::statusUpdated, info, [info] (IdmInfo *idmInfo) {
+            if (idmInfo->connected) {
+                info->finish(Thing::ThingErrorNoError);
+            }
+        });
+        connect(idm, &Idm::destroyed, this, [thing, this] {m_idmConnections.remove(thing);});
+        connect(info, &ThingSetupInfo::aborted, idm, &Idm::deleteLater);
+        connect(idm, &Idm::statusUpdated, this, &IntegrationPluginIdm::onStatusUpdated);
+
+    } else {
+        Q_ASSERT_X(false, "setupThing", QString("Unhandled thingClassId: %1").arg(thing->thingClassId().toString()).toUtf8());
+    }
+}
+
+void IntegrationPluginIdm::postSetupThing(Thing *thing)
+{
+    qCDebug(dcIdm()) << "postSetupThing called" << thing->name();
+
+    if (!m_refreshTimer) {
+        m_refreshTimer = hardwareManager()->pluginTimerManager()->registerTimer(10);
+        connect(m_refreshTimer, &PluginTimer::timeout, this, &IntegrationPluginIdm::onRefreshTimer);
+    }
+
+    if (thing->thingClassId() == navigator2ThingClassId) {
+        qCDebug(dcIdm()) << "Thing id: " << thing->id();
+        Idm *idm = m_idmConnections.value(thing);
+
+        if (idm != nullptr) {
+
+            qCDebug(dcIdm()) << "Thing set up, calling update";
+            update(thing);
+
+            thing->setStateValue(navigator2ConnectedStateTypeId, true);
+        }
+    }
+}
+
+void IntegrationPluginIdm::thingRemoved(Thing *thing)
+{
+    qCDebug(dcIdm()) << "thingRemoved called" << thing->name();
+
+    if (thing->thingClassId() == navigator2ThingClassId) {
+        if (m_idmConnections.contains(thing)) {
+            m_idmConnections.take(thing)->deleteLater();
+        }
+    }
+}
+
+void IntegrationPluginIdm::executeAction(ThingActionInfo *info)
+{
+    Thing *thing = info->thing();
+    Action action = info->action();
+
+    if (thing->thingClassId() == navigator2ThingClassId) {
+        if (action.actionTypeId() == navigator2PowerActionTypeId) {
+
+        } else {
+            Q_ASSERT_X(false, "executeAction", QString("Unhandled action: %1").arg(action.actionTypeId().toString()).toUtf8());
+        }
+    } else {
+        Q_ASSERT_X(false, "executeAction", QString("Unhandled thingClassId: %1").arg(thing->thingClassId().toString()).toUtf8());
+    }
+}
+
+void IntegrationPluginIdm::update(Thing *thing)
+{
+    if (thing->thingClassId() == navigator2ThingClassId) {
+        qCDebug(dcIdm()) << "Updating thing";
+
+        Idm *idm = m_idmConnections.value(thing);
+
+        if (idm != nullptr) {
+            idm->onRequestStatus();
+        }
+    }
+}
+
+void IntegrationPluginIdm::onStatusUpdated(IdmInfo *info)
+{
+    /* qCDebug(dcIdm()) << "onStatusUpdated"; */
```

Comment:
I left some notes in the Idm class regarding the info. If you clean it up, do not delete it here, rather than in the class where you allocate the memory.